platform/src/main/java/org/hillview/utils/IntSet.java

```diff
@@ -94,20 +98,21 @@ public boolean add(final int k) {
         if (this.size++ >= this.maxFill) {
             this.rehash(HashUtil.arraySize(this.size + 1, this.f));
         }
+        this.iteratorKey = this.key;
         return true;
     }
 
     /**
      *
      * @param pos a location in the key[] array
-     * @return the location of the next full slot after cursor.
+     * @return the location of the next full slot after cursor. Operates on the iteratorKey array
      */
     public int getNext(int pos) {
-        while ( key[pos & this.mask] == 0) { pos++; }
+        while ( iteratorKey[pos & this.mask] == 0) { pos++; }
```

Comment:
this.
Previously the code was copy-and-paste, but if you edit it you should also use our coding conventions.